rfcs/003-strings.md

```diff
@@ -0,0 +1,325 @@
+# Strings in windows-rs
+
+## Summary
+
+Rust and Windows each have a fair amount of string types. This RFC proposes a system for making it easy and obvious how to make the string types from both environments work well together.
+
+## Motivation
+
+Making Rust and Windows string types work together is integral to ensuring correct, performant, and boiler-plate free code.
+
+### What we want to accomplish
+
+In general, we are trying to accomplish two goals:
+
+* Correctly model the Windows string types in a way that leverages Rust's type and ownership systems to prevent incorrect use.
+* Allow for easy and low cost ways to convert between Windows and Rust types.
+
+On top of this, we want these APIs to feel natural and obvious to Rust developers.
+
+## Explanation
+
+### The Different Types of Strings
+
+#### Windows
+
+Windows has the following string types:
+
+* `PSTR`: a mutable, nul-terminated string (i.e., composed of "characters" a.k.a. `u8`s) that is often but not always ANSI encoded.
+* `PCSTR`: an immutable version of `PSTR`.
+  * **QUESTION**: what are the practical differences between this and `PSTR`?
```

Comment:
The difference is that `PCSTR` can be created from a `CStr` reference, and can also be converted back.